File: api/src/endpoint/task/get_task.rs
use crate::{endpoint::Endpoint, resource::TaskResource};

pub struct GetTask;

impl Endpoint for GetTask {
    const PATH: &'static str = "/ci/task/{id}";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = GetTaskResponse;
}

pub type GetTaskResponse = TaskResource;

File: gitdot-api-ts/src/endpoint/task/get-task.ts
import { z } from "zod";
import { TaskResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetTaskRequest = z.object({});
export type GetTaskRequest = z.infer<typeof GetTaskRequest>;

export const GetTaskResponse = TaskResource;
export type GetTaskResponse = z.infer<typeof GetTaskResponse>;

export const GetTask = {
  path: "/ci/task/{id}",
  method: "GET",
  request: GetTaskRequest,
  response: GetTaskResponse,
} as const satisfies Endpoint;
export type GetTask = typeof GetTask;
